### PR TITLE
rpc: decouple test fixtures from node implementation

### PR DESCRIPTION
**Pull request by tychoish**
I've been detangling bits of the rpc tests for a bit, and this avoids
direct dependencies (except in one minor case) in the exposed
interface of the rpc test package relative to the node package.

By removing this dependency in the tests, it makes most of the other
node dependency changes much eaiser to think about and reason about.

### CODE REVIEW

File: libs/service/service.go [hunk 1]
```diff
@@ -50,12 +50,15 @@ type Service interface {
...
+
+	// Wait blocks until the service is stopped.
+	Wait()
 }
 
 /*
 Classical-inheritance-style service declarations. Services can be started, then
 stopped, then optionally restarted.
-
+k
```
Comment (alexanderbez) [inline libs/service/service.go:61]: Que?

Comment (tychoish) [reply libs/service/service.go:61]: missstakeeeee

**Comment by tychoish** [comment]
There are a couple of flakey tests still, but I don't think there are needed/pending changes to this PR at this point. Let me know if there's anything more that stands out to you all.

File: light/provider/http/http_test.go [hunk 2]
```diff
@@ -33,23 +33,28 @@ func TestNewProvider(t *testing.T) {
 	require.Equal(t, fmt.Sprintf("%s", c), "http{http://153.200.0.1}")
 }
 
-func NodeSuite(t *testing.T) *node.Node {
+func NodeSuite(t *testing.T) (service.Service, *config.Config) {
```
Comment (cmwaters) [inline light/provider/http/http_test.go:36]:
```suggestion
// NodeSuite initiates and runs a full node instance in the background, stopping 
// it once the test is completed
func NodeSuite(t *testing.T) (service.Service, *config.Config) {
```

Comment (cmwaters) [reply light/provider/http/http_test.go:36]: Do you think there would ever be an instance where the test itself wants to pass over the context or is it safe to just always create the context here

File: rpc/grpc/grpc_test.go [hunk 3]
```diff
@@ -7,27 +7,36 @@ import (
...
 	"github.com/tendermint/tendermint/abci/example/kvstore"
-	"github.com/tendermint/tendermint/node"
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/service"
 	core_grpc "github.com/tendermint/tendermint/rpc/grpc"
 	rpctest "github.com/tendermint/tendermint/rpc/test"
 )
 
-func NodeSuite(t *testing.T) *node.Node {
+func NodeSuite(t *testing.T) (service.Service, *config.Config) {
```
Comment (cmwaters) [inline rpc/grpc/grpc_test.go:16]: Seeing as this function gets repeated a couple of times, why not add it to `rpc/test/helpers.go`

File: rpc/grpc/grpc_test.go [hunk 4]
```diff
@@ -7,27 +7,36 @@ import (
...
 	t.Helper()
 
+	ctx, cancel := context.WithCancel(context.Background())
+
+	conf := rpctest.CreateConfig()
+
 	// start a tendermint node in the background to test against
 	app := kvstore.NewApplication()
-	node := rpctest.StartTendermint(app)
+	app.RetainBlocks = 9
```
Comment (cmwaters) [inline rpc/grpc/grpc_test.go:25]: What's the reasoning for this?

File: rpc/test/helpers.go [hunk 5]
```diff
@@ -106,74 +100,65 @@ func GetGRPCClient(conf *cfg.Config) core_grpc.BroadcastAPIClient {
...
+	waitForRPC(ctx, conf)
+	waitForGRPC(ctx, conf)
+
+	if !nodeOpts.suppressStdout {
+		fmt.Println("Tendermint running!")
+	}
+
+	return node, func(ctx context.Context) error {
+		if err := node.Stop(); err != nil {
+			logger.Error("Error when tryint to stop node", "err", err)
```
Comment (cmwaters) [inline rpc/test/helpers.go:156]:
```suggestion
			logger.Error("Error when trying to stop node", "err", err)
```

**Review by cmwaters — APPROVED** [review]
👍

File: light/provider/http/http_test.go (hunk 2 above)
Comment (tychoish) [reply light/provider/http/http_test.go:36]: I think probably we don't want to be creating contexts here and it'd be better to pass it in, but given how we're using contexts at the moment, I think it's safe to increment toward avoiding creating a context in this case.

File: rpc/grpc/grpc_test.go (hunk 4 above)
Comment (tychoish) [reply rpc/grpc/grpc_test.go:25]: copypasta, good catch

File: rpc/grpc/grpc_test.go (hunk 3 above)
Comment (tychoish) [reply rpc/grpc/grpc_test.go:16]: the rpc/client version uses a persistentkv application and the others don't. I'm inclined to leave it the way it is now and clean it up later as part of more/deeper thinking we should do about the RPC layer

**Pull request merged**